src/Futhark/Internalise/NormaliseSize.hs

```diff
@@ -0,0 +1,536 @@
+-- | This Normalisation module converts a well-typed, polymorphic,
+-- module-free Futhark program into an equivalent program without
+-- arbitrary size expression.
```

Comment:
Please put a comment indicating the overall strategy and important terms (e.g. "intros").